test/Strings/lastindexof.js

```diff
@@ -27,6 +27,9 @@ WScript.Echo(str2.lastIndexOf("\0ab"));
 var str3 = "\u0100\u0111\u0112\u0113";
 WScript.Echo(str3.lastIndexOf("\u0112\u0113"));
 
+var str4 = "\u0061\u0062\u0063\u3042\u3044";
+WScript.Echo(str4.lastIndexOf("\u3042\u3044", 1));

```

Comment:
It doesn't look like we have existing test coverage for this case and, therefore, I think we should add more tests. The ones I have in mind are:

  * `str4.lastIndexOf("\u3042\u3044") === 3`: Successful match without `position`
  * `str4.lastIndexOf("\u3042\u3044", 4) === 3`: Successful match ending at `position`
  * `"\u3042\u3044\u0061".lastIndexOf("\u3042\u3044", 3) === 0`: Successful match ending before `position`
  * `str4.lastIndexOf("\u3044\0", 5) === -1`: Verify that we don't match `"\0"` at the end of the string in C++